SCSMA 2016 Support

mortenlerudjordet opened the pull request:
Div improvements to logging and added support for port parameter.
Initial create is a long time ago, but as there are activity here now I will PR my changes.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xscsma/23)
<!-- Reviewable:end -->

Commit:
Improvements

Commit:
Converted tabs to space

johlju commented:
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[DSCResources/MSFT_xSCSMAPowerShellSetup/MSFT_xSCSMAPowerShellSetup.psm1, line 45 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD276Dm3PcPu9mGNc5H:-LD276Dm3PcPu9mGNc5I:b-9ao6tl) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMAPowerShellSetup/MSFT_xSCSMAPowerShellSetup.psm1#L45)):*
> ```PowerShell
>             $IdentifyingNumber = "{EF2760C1-FED5-45FD-B067-D9419F7DEBEF}"
>         }
>         "7.3.149.0"
> ```

Can you confirm that this is correct build number?

---

*[DSCResources/MSFT_xSCSMAPowerShellSetup/MSFT_xSCSMAPowerShellSetup.psm1, line 120 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD27CBv6MRxSSSMPh85:-LD27CBv6MRxSSSMPh86:b-9ao6tl) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMAPowerShellSetup/MSFT_xSCSMAPowerShellSetup.psm1#L120)):*
> ```PowerShell
>             $IdentifyingNumber = "{EF2760C1-FED5-45FD-B067-D9419F7DEBEF}"
>         }
>         "7.3.149.0"
> ```

Can you confirm that this is correct build number?

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 54 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD2A4Qc4AVyEiRbWSQ8:-LD2A4Qc4AVyEiRbWSQ9:b771gjk) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L54)):*
> ```PowerShell
> 
>         [System.String]
>         $LogMSIinstall = $False,
> ```

We should not have unused non-mandatory parameters in the Get-TargetResource.
https://github.com/PowerShell/DscResources/blob/master/BestPractices.md#get-targetresource-should-not-contain-unused-non-mandatory-parameters

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 184 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD2A2dLEhSAci-Sdyto:-LD2A2dLEhSAci-Sdytp:b54q0f4) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L184)):*
> ```
> $False
> ```

$false

Throughout.

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 291 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD2AeGz135r4rU23M_n:-LD2AeGz135r4rU23M_o:b-eckgaa) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L291)):*
> ```PowerShell
> 
>             # Check if logging is wanted
>             if($LogMSIinstall) {
> ```

Could we make a helper function of this that can be reused by both the updated resources?

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 295 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD29PzP31o5AmSvNYCy:-LD29PzP31o5AmSvNYCz:b-ha8moh) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L295)):*
> ```
> If(
> ```

if ( (lower-case 'i')

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 299 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD29UEb2PjPq9IbLc89:-LD29UEb2PjPq9IbLc8A:b-8ufepp) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L299)):*
> ```
> Else
> ```

else (lower-case 'e')

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 299 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD29XTZ4msh979BGG4M:-LD29ROk7xWlEhihqmgO:b9o3b0c) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L299)):*
> ```PowerShell
>                     New-Item -ItemType Directory -Force -Path $MSIlogPath
>                 }
>                 Else {
> ```

We should have open brace on a separate row.

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 301 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD29jpRBWfTKlKeuNJz:-LD29jpRBWfTKlKeuNK-:bwohr5y) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L301)):*
> ```
> exsists
> ```

exists

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof, line 18 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD28duLD773W_jpwC_S:-LD28duLD773W_jpwC_T:btr8kmu) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof#L18)):*
> ```
>     [Write, Description("Use Microsoft Update.")] String MSUpdate;
>     [Write, Description("Product key for licensed installations.")] String ProductKey;
>     [Write, Description("Set to activate logging during install of MSI.")] String LogMSIinstall;
> ```

Please add these properties to the resource parameters in the README.md, please add the default values to the description. Make sure the same description is here and in the README.md.

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1, line 79 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD2APy-81Ru_A8lsYKo:-LD2APy-81Ru_A8lsYKp:b771gjk) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1#L79)):*
> ```PowerShell
> 
>         [System.String]
>         $LogMSIinstall = $False,
> ```

We should not have unused non-mandatory parameters in the Get-TargetResource.
https://github.com/PowerShell/DscResources/blob/master/BestPractices.md#get-targetresource-should-not-contain-unused-non-mandatory-parameters

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1, line 251 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD2ARh_5HdLXOOMWNbv:-LD2ARh_5HdLXOOMWNbw:b54q0f4) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1#L251)):*
> ```
> $False
> ```

$false

Throughout.

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof, line 26 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD291aN0vThGgqfeTML:-LD291aN0vThGgqfeTMM:btr8kmu) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof#L26)):*
> ```
>     [Write, Description("Product key for licensed installations.")] String ProductKey;
>     [Write, Description("Array of Runbook Worker servers in this deployment.")] String RunbookWorkerServers[];
>     [Write, Description("Set to activate logging during install of MSI.")] String LogMSIinstall;
> ```

Please add these properties to the resource parameters in the README.md, please add the default values to the description. Make sure the same description is here and in the README.md.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23)*
<!-- Sent from Reviewable.io -->

Commit:
Fixes

mortenlerudjordet commented:
I had forgotten to include the changes I did to README, I did not add my change note to this one. They are:
* Added MSI install logging
* Fixed $IdentifyingNumber for System Center 2016 PS module install
* Added missing -Port parameter argument when using SMA PS modules during install

I also included the script I used to test install as an example. 

When I was testing this way back these where the correct number for the SMA binaries. 

The MSI logging was needed during testing to be able to troubleshoot, I did not remove it as I though others might benefit. I have changed for it to be an mandatory value. Could have been done better but this was what I had time to do back then.

Commit:
Fixed metadata for LogMSIInstall

Commit:
Removed tab

Commit:
Removed tab

Commit:
Removed tab again

Commit:
fixed

mortenlerudjordet commented:
Review status: 1 of 8 files reviewed at latest revision, 13 unresolved discussions.

---

*[DSCResources/MSFT_xSCSMAPowerShellSetup/MSFT_xSCSMAPowerShellSetup.psm1, line 45 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD276Dm3PcPu9mGNc5H:-LD2xewbDxaoeeAbVKSF:bkc0zae) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMAPowerShellSetup/MSFT_xSCSMAPowerShellSetup.psm1#L45)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Can you confirm that this is correct build number?
</blockquote></details>

It was when I did my testing over a year ago

---

*[DSCResources/MSFT_xSCSMAPowerShellSetup/MSFT_xSCSMAPowerShellSetup.psm1, line 120 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD27CBv6MRxSSSMPh85:-LD2xvD1A8UyTfl0k5PO:b-mb2j28) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMAPowerShellSetup/MSFT_xSCSMAPowerShellSetup.psm1#L120)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Can you confirm that this is correct build number?
</blockquote></details>

It was the correct number when I did my testing over a year ago

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 54 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD2A4Qc4AVyEiRbWSQ8:-LD2y0rN-gQiZS6ixpuu:bhdy97) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L54)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

We should not have unused non-mandatory parameters in the Get-TargetResource.
https://github.com/PowerShell/DscResources/blob/master/BestPractices.md#get-targetresource-should-not-contain-unused-non-mandatory-parameters
</blockquote></details>

Have changed it to be mandatory

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 184 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD2A2dLEhSAci-Sdyto:-LD2y5Wq2Q_MDKpVaRVh:b-886gsx) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L184)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> $False
> ```

$false

Throughout.
</blockquote></details>

Fixed

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 291 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD2AeGz135r4rU23M_n:-LD2y8lF8kIHIyxTLte3:b-xybftb) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L291)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Could we make a helper function of this that can be reused by both the updated resources?
</blockquote></details>

Sound like a good idea, I did not have much time when I did this. And have moved on to other technologies since.

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 295 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD29PzP31o5AmSvNYCy:-LD2yJHS-s49wpEL_Suc:b-886gsx) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L295)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> If(
> ```

if ( (lower-case 'i')
</blockquote></details>

Fixed

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23)*
<!-- Sent from Reviewable.io -->

johlju commented:
Review status: 1 of 8 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 54 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD2A4Qc4AVyEiRbWSQ8:-LD5og_x69LySS41cff2:be8jqie) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L54)):*
<details><summary><i>Previously, mortenlerudjordet (Morten) wrote…</i></summary><blockquote>

Have changed it to be mandatory
</blockquote></details>

I would rather have it non-mandatory with a default value of $false, otherwise this is a breaking change. Do you see it must be mandatory? 

My initial comment was just to remove the parameters from the Get-TargetResource function if the were unused in the Get-TargetResource function. :slightly_smiling_face: 

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 303 at r3](https://reviewable.io/reviews/powershell/xscsma/23#-LD5pIOc4OK1IWwe8Zjn:-LD5pIOc4OK1IWwe8Zjo:b-3ib4vc) ([raw file](https://github.com/powershell/xscsma/blob/7dadc9a5ede90b4a17be2a10fbe34b81f5f1f1b0/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L303)):*
> ```PowerShell
>                 else {
>                     if(Test-Path -Path $logPathName) 
> 					{
> ```

There are tabs on this row. can we convert them to spaces?

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1, line 408 at r3](https://reviewable.io/reviews/powershell/xscsma/23#-LD5pSVg7bHjFz_Xigf4:-LD5pSVg7bHjFz_Xigf5:b-3ib4vc) ([raw file](https://github.com/powershell/xscsma/blob/7dadc9a5ede90b4a17be2a10fbe34b81f5f1f1b0/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1#L408)):*
> ```PowerShell
>                 else {
>                     if(Test-Path -Path $logPathName) 
> 					{
> ```

There are tabs on this row. can we convert them to spaces?

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23)*
<!-- Sent from Reviewable.io -->

Commit:
Revert MSI logging mandatory

Fixed tab issue again

Commit:
Revert logging

mortenlerudjordet commented:
Thanks, learned something new.

mortenlerudjordet commented:
Review status: 1 of 8 files reviewed at latest revision, 10 unresolved discussions.

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 54 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD2A4Qc4AVyEiRbWSQ8:-LDCQ3pcA6mChOfK25-_:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L54)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

I would rather have it non-mandatory with a default value of $false, otherwise this is a breaking change. Do you see it must be mandatory? 

My initial comment was just to remove the parameters from the Get-TargetResource function if the were unused in the Get-TargetResource function. :slightly_smiling_face: 
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 299 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD29UEb2PjPq9IbLc89:-LDCPzc74pE86CgBxyPi:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L299)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> Else
> ```

else (lower-case 'e')
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 299 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD29XTZ4msh979BGG4M:-LDCPzwcEdA-H8e-BCcQ:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L299)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

We should have open brace on a separate row.
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 301 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD29jpRBWfTKlKeuNJz:-LDCQ-_L6o6PzJ0YBaO1:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L301)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> exsists
> ```

exists
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 303 at r3](https://reviewable.io/reviews/powershell/xscsma/23#-LD5pIOc4OK1IWwe8Zjn:-LDCQ-JNCfBYQ8z4yRmx:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/7dadc9a5ede90b4a17be2a10fbe34b81f5f1f1b0/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L303)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

There are tabs on this row. can we convert them to spaces?
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof, line 18 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD28duLD773W_jpwC_S:-LDCQ0GTBAhEJeusnj1z:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof#L18)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please add these properties to the resource parameters in the README.md, please add the default values to the description. Make sure the same description is here and in the README.md.
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1, line 79 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD2APy-81Ru_A8lsYKo:-LDCQ0ps6Km6Z2thbc14:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1#L79)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

We should not have unused non-mandatory parameters in the Get-TargetResource.
https://github.com/PowerShell/DscResources/blob/master/BestPractices.md#get-targetresource-should-not-contain-unused-non-mandatory-parameters
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1, line 251 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD2ARh_5HdLXOOMWNbv:-LDCQ15i6fmjh-KjXSZ-:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1#L251)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> $False
> ```

$false

Throughout.
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1, line 408 at r3](https://reviewable.io/reviews/powershell/xscsma/23#-LD5pSVg7bHjFz_Xigf4:-LDCQ1zvEmMwar0CuxBo:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/7dadc9a5ede90b4a17be2a10fbe34b81f5f1f1b0/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1#L408)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

There are tabs on this row. can we convert them to spaces?
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof, line 26 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD291aN0vThGgqfeTML:-LDCQ2S8AOYGRQTM1cAg:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof#L26)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please add these properties to the resource parameters in the README.md, please add the default values to the description. Make sure the same description is here and in the README.md.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23)*
<!-- Sent from Reviewable.io -->

mortenlerudjordet commented:
First time using reviewable so some learning curve expected

mortenlerudjordet commented:
Review status: 1 of 8 files reviewed at latest revision, 10 unresolved discussions.

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1, line 408 at r3](https://reviewable.io/reviews/powershell/xscsma/23#-LD5pSVg7bHjFz_Xigf4:-LDCQapj2ZCShzMrwE7g:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/7dadc9a5ede90b4a17be2a10fbe34b81f5f1f1b0/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1#L408)):*
<details><summary><i>Previously, mortenlerudjordet (Morten) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof, line 26 at r1](https://reviewable.io/reviews/powershell/xscsma/23#-LD291aN0vThGgqfeTML:-LDCQbUGDrCm25MTb40G:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/b229e5ae204c398e8e5578547f930c9216365798/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof#L26)):*
<details><summary><i>Previously, mortenlerudjordet (Morten) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23)*
<!-- Sent from Reviewable.io -->

mortenlerudjordet commented:
Reviewed 1 of 5 files at r1, 2 of 5 files at r2, 4 of 4 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewed 2 of 5 files at r2, 2 of 4 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---

*[README.md, line 59 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH46oY-LTZOBnQhhyq:-LDH46oY-LTZOBnQhhyr:btbnbn6) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L59)):*
> ```
> LogMSIinstall
> ```

We should write this as `LogMsiInstall` (see upper and lower-case)

---

*[README.md, line 60 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH49-j4saEZ-uxHdkU:-LDH49-j4saEZ-uxHdkV:bzcie9u) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L60)):*
> ```
> MSIlogPath
> ```

We should write this as `MsiLogPath` (see upper and lower-case)

---

*[README.md, line 83 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH441O-w3KBkLzcMB8:-LDH441O-w3KBkLzcMB9:btbnbn6) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L83)):*
> ```
> LogMSIinstall
> ```

We should write this as `LogMsiInstall` (see upper and lower-case)

---

*[README.md, line 84 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH49VWBElAGtMOZ5j0:-LDH49VWBElAGtMOZ5j1:bzcie9u) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L84)):*
> ```
> MSIlogPath
> ```

We should write this as `MsiLogPath` (see upper and lower-case)

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 290 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH3TOW5k2BwynX-Ftx:-LDH3TOW5k2BwynX-Fty:b5q0bn2) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L290)):*
> ```
> else {
> ```

We should have open brace on a separate row. 

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof, line 18 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH3qNy-e2rIMEn5brQ:-LDH3qNy-e2rIMEn5brR:btbnbn6) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof#L18)):*
> ```
> LogMSIinstall
> ```

We should write this as `LogMsiInstall` (see upper and lower-case)

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof, line 19 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH3xGNAl79jXVVjDJm:-LDH3xGNAl79jXVVjDJn:bzcie9u) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof#L19)):*
> ```
> MSIlogPath
> ```

We should write this as `MsiLogPath` (see upper and lower-case)

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1, line 395 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH4KWLAPf-uIJRcpMK:-LDH4KWLAPf-uIJRcpML:b5q0bn2) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1#L395)):*
> ```
> else {
> ```

We should have open brace on a separate row. 

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof, line 26 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH4PmzCCwYU9_Fa1y3:-LDH4PmzCCwYU9_Fa1y4:btbnbn6) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof#L26)):*
> ```
> LogMSIinstall
> ```

We should write this as `LogMsiInstall` (see upper and lower-case)

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof, line 27 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH4RJ-A_MvOsQNWQaL:-LDH4RJ-A_MvOsQNWQaM:bzcie9u) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof#L27)):*
> ```
> MSIlogPath
> ```

We should write this as `MsiLogPath` (see upper and lower-case)

---

*[Examples/SCSMA-PreInstalledSQL.ps1, line 48 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH4k966HmR_qhikkQz:-LDH4k966HmR_qhikkR-:bt7nmby) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/Examples/SCSMA-PreInstalledSQL.ps1#L48)):*
> ```PowerShell
> Configuration SMA
> {
>     #Import-DscResource -Module xSQLServer
> ```

This should be commented?

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23)*
<!-- Sent from Reviewable.io -->

Commit:
Syntax fixes

mortenlerudjordet commented:
Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[README.md, line 59 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH46oY-LTZOBnQhhyq:-LDHaPcWD8GyK_2YNTf0:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L59)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> LogMSIinstall
> ```

We should write this as `LogMsiInstall` (see upper and lower-case)
</blockquote></details>

Done.

---

*[README.md, line 60 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH49-j4saEZ-uxHdkU:-LDHaQ8cELfcKFhVeRrJ:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L60)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> MSIlogPath
> ```

We should write this as `MsiLogPath` (see upper and lower-case)
</blockquote></details>

Done.

---

*[README.md, line 83 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH441O-w3KBkLzcMB8:-LDHaQzg-JvVMdee47A9:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L83)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> LogMSIinstall
> ```

We should write this as `LogMsiInstall` (see upper and lower-case)
</blockquote></details>

Done.

---

*[README.md, line 84 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH49VWBElAGtMOZ5j0:-LDHaRLLAjRYrRKN407b:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L84)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> MSIlogPath
> ```

We should write this as `MsiLogPath` (see upper and lower-case)
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 290 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH3TOW5k2BwynX-Ftx:-LDHaSw585uirwaHURgm:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L290)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> else {
> ```

We should have open brace on a separate row. 
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof, line 18 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH3qNy-e2rIMEn5brQ:-LDHaUW4BhJU4ONB1rlh:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof#L18)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> LogMSIinstall
> ```

We should write this as `LogMsiInstall` (see upper and lower-case)
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof, line 19 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH3xGNAl79jXVVjDJm:-LDHaV-g2mrdH9ViXF6W:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof#L19)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> MSIlogPath
> ```

We should write this as `MsiLogPath` (see upper and lower-case)
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1, line 395 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH4KWLAPf-uIJRcpMK:-LDHaVsID77hGOkRCM4C:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1#L395)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> else {
> ```

We should have open brace on a separate row. 
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof, line 26 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH4PmzCCwYU9_Fa1y3:-LDHaWpy0s4el5ywJEWm:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof#L26)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> LogMSIinstall
> ```

We should write this as `LogMsiInstall` (see upper and lower-case)
</blockquote></details>

Done.

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof, line 27 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH4RJ-A_MvOsQNWQaL:-LDHaXCR8lP-m8_SHoz8:b-896fix) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof#L27)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> MSIlogPath
> ```

We should write this as `MsiLogPath` (see upper and lower-case)
</blockquote></details>

Done.

---

*[Examples/SCSMA-PreInstalledSQL.ps1, line 48 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH4k966HmR_qhikkQz:-LDHaYIO7lZGvh0xw0Nm:b-5hx6na) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/Examples/SCSMA-PreInstalledSQL.ps1#L48)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This should be commented?
</blockquote></details>

Added comment about adding this in again if module is used for creating SQL instance

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23)*
<!-- Sent from Reviewable.io -->

Commit:
Syntax fixes

johlju commented:
You doing great with Reviewable! :slightly_smiling_face:  For some reasone I couldn't acknowledge some of the comments that was done. Not your fault, think it's a reported bug in Reviewable. Please write 'Done' without the full stop `.` and I see if I can acknowledge them.

Not seeing some changes. Maybe you missed to push? 

---

Reviewed 3 of 3 files at r6, 2 of 2 files at r7.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[README.md, line 59 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH46oY-LTZOBnQhhyq:-LDMb7Uz-um4zNof7R5h:bzgj0fn) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L59)):*
<details><summary><i>Previously, mortenlerudjordet (Morten) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Not seeing this change yet.

---

*[README.md, line 60 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH49-j4saEZ-uxHdkU:-LDMbB0o2keutDND8B_l:b-u0w4m9) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L60)):*
> Not seeing this change yet.
Not seeing this change yet.

---

*[README.md, line 83 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH441O-w3KBkLzcMB8:-LDMbCp39OyWm7r8hQ-D:bzgj0fn) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L83)):*
<details><summary><i>Previously, mortenlerudjordet (Morten) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Not seeing this change yet.

---

*[README.md, line 84 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH49VWBElAGtMOZ5j0:-LDMbDjg6-TrbpPGsu1N:bzgj0fn) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L84)):*
<details><summary><i>Previously, mortenlerudjordet (Morten) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Not seeing this change yet.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23#-:-LDMbn2XD-UVGxppdDi_:blw3wwx)*
<!-- Sent from Reviewable.io -->

Commit:
MSI* syntax fix again

mortenlerudjordet commented:
Review status: 7 of 8 files reviewed at latest revision, 10 unresolved discussions.

---

*[README.md, line 60 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH49-j4saEZ-uxHdkU:-LDMnIqz1KbXaP5u0Dz1:bb6un1j) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L60)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> Not seeing this change yet.
Not seeing this change yet.
</blockquote></details>

Done

---

*[README.md, line 83 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH441O-w3KBkLzcMB8:-LDMnM11BfSvPqQ5gmG3:bb6un1j) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L83)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Not seeing this change yet.
</blockquote></details>

Done

---

*[README.md, line 84 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH49VWBElAGtMOZ5j0:-LDMnOh_13nDtBWv1Zu-:bb6un1j) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L84)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Not seeing this change yet.
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23)*
<!-- Sent from Reviewable.io -->

mortenlerudjordet commented:
I find it a bit confusing to navigate, though iterations are my friend in all things. So it will sink in eventually. I've gone and done Done again all over the place.  

---

Review status: 7 of 8 files reviewed at latest revision, 10 unresolved discussions.

---

*[README.md, line 59 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH46oY-LTZOBnQhhyq:-LDMnxKT5V_fhiJ8hB5u:bb6un1j) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/README.md#L59)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Not seeing this change yet.
</blockquote></details>

Done

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1, line 290 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH3TOW5k2BwynX-Ftx:-LDMo-8tAWTE3drham_O:bb6un1j) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.psm1#L290)):*
<details><summary><i>Previously, mortenlerudjordet (Morten) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof, line 18 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH3qNy-e2rIMEn5brQ:-LDMo13SEzThduZ26wQ7:bb6un1j) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof#L18)):*
<details><summary><i>Previously, mortenlerudjordet (Morten) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done

---

*[DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof, line 19 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH3xGNAl79jXVVjDJm:-LDMo1nJ6fX7jBObrPnT:bb6un1j) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMARunbookWorkerServerSetup/MSFT_xSCSMARunbookWorkerServerSetup.schema.mof#L19)):*
<details><summary><i>Previously, mortenlerudjordet (Morten) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1, line 395 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH4KWLAPf-uIJRcpMK:-LDMo2x47Px1XKnyr_Xr:bb6un1j) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.psm1#L395)):*
<details><summary><i>Previously, mortenlerudjordet (Morten) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof, line 26 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH4PmzCCwYU9_Fa1y3:-LDMo3os3KtGimywguou:bb6un1j) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof#L26)):*
<details><summary><i>Previously, mortenlerudjordet (Morten) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done

---

*[DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof, line 27 at r5](https://reviewable.io/reviews/powershell/xscsma/23#-LDH4RJ-A_MvOsQNWQaL:-LDMo4b-6OzeS3Kvli8y:bb6un1j) ([raw file](https://github.com/powershell/xscsma/blob/efd5a8d354adbf9f73295aeda677ff37a8201a10/DSCResources/MSFT_xSCSMAWebServiceServerSetup/MSFT_xSCSMAWebServiceServerSetup.schema.mof#L27)):*
<details><summary><i>Previously, mortenlerudjordet (Morten) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23#-:-LDMnYN4ELDwB61QI6IP:b-og3x1c)*
<!-- Sent from Reviewable.io -->

mortenlerudjordet commented:
Reviewed 2 of 2 files at r7, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewable can be overwhelming at first, but powerful when you get used to it. For me it helped a lot by pressing 'h' to get help. 

Reviewable just released (this weekend) a new version of the discussion/blocking semantics so that should work better now too. :slightly_smiling_face: 

---

Review status: all files reviewed at latest revision, all discussions resolved.

---

*[a discussion](https://reviewable.io/reviews/powershell/xscsma/23#-LDb3lsS6t3gl6ukjX1x:-LDb3lsS6t3gl6ukjX1y:b3wkxpy) (no related file):*
Would you mind adding unit tests for, at least, the changes you made here?

Template here; https://github.com/PowerShell/DscResources/blob/master/Tests.Template/unit_template.ps1

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23#-:-LDb2uU-3zMhcgNomXVP:b995pav)*
<!-- Sent from Reviewable.io -->

johlju commented:
@mortenlerudjordet Please rebase this PR since a PR was recently merged.

mortenlerudjordet commented:
I have not much experience in writing tests, so I will need to learn before being able to execute on this. Also  it will take some extra time to merge the changes already done into mine. I am not a developer so these things are not something I do every day.

johlju commented:
Since there are no tests for this resource yet, I can merge these changes without tests, but I wanted to ask before making a exception.

You can read here how to [resolve merge conflicts](https://github.com/PowerShell/DscResources/blob/master/GettingStartedWithGitHub.md#resolve-merge-conflicts). Ping me if you having any problem and I will help.

mortenlerudjordet commented:
Learning how to write tests is on my TODO list, though it would have gone faster if I could have had something to compare against. As there are none in this project, I feel the need for deeper understanding of good practices before taking on the task. 

I will resolve the conflicts this weekend.

johlju commented:
It's a good skill to have, I promise you will have use for it, not just testing code, but for testing anything. 

Suggest reading [The Pester Book](https://leanpub.com/pesterbook) that I personally thought was a very good read.

If you want to learn by trial-and-error, here is a good example to take after.
**Unit test:** https://github.com/PowerShell/SqlServerDsc/blob/dev/Tests/Unit/MSFT_SqlServerDatabaseMail.Tests.ps1
**Integration test:** https://github.com/PowerShell/SqlServerDsc/blob/dev/Tests/Integration/MSFT_SqlServerDatabaseMail.Integration.Tests.ps1
**Integration test configuration (dot-source into the previous file):** 
https://github.com/PowerShell/SqlServerDsc/blob/dev/Tests/Integration/MSFT_SqlServerDatabaseMail.config.ps1

If you want to give it a try, I'm happy give you pointers if you get stuck. 🙂

johlju commented:
It's enough with a unit test. A unit test mocks everything (cmdlets) and does not actually make any changes to the system, and a unit test should be able to run on any client without having any bits installed. 
Integration tests actually tests by performing the actions and verifies the result. 

I just added links to the integration tests for reference, we don't have to make an integration tests for this PR.

Commit:
Merge

mortenlerudjordet commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io/reviews/powershell/xscsma/23#-LDb3lsS6t3gl6ukjX1x:-LDvyyDsD0-GWk3SUmHW:b-dzf141) (no related file):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Would you mind adding unit tests for, at least, the changes you made here?

Template here; https://github.com/PowerShell/DscResources/blob/master/Tests.Template/unit_template.ps1
</blockquote></details>

If I figure out the test part I can do a new PR later if that works?

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23)*
<!-- Sent from Reviewable.io -->

mortenlerudjordet commented:
Did the merge, VS code gave me some lip but I think it's sorted now.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23#-:-LDvz4TP3ESYWZpC3h8E:beyt7mf)*
<!-- Sent from Reviewable.io -->

johlju commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

I make an exception with the unit tests. Looking forward to a test later. :slightly_smiling_face: 

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23#-:-LE9hfKB4ehqTPSDblwB:b-hog4ns)*
<!-- Sent from Reviewable.io -->

johlju commented:
Review status: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---

*[a discussion](https://reviewable.io/reviews/powershell/xscsma/23#-LE9iEVo07vkcS-vcBM5:-LE9iEVo07vkcS-vcBM6:b1i4k5d) (no related file):*
@mortenlerudjordet I missed that the Unreleased section wasn't updated. 

Could you please add an descriptive entry, for each change/issue, to the Unreleased section of the change log in the file README.md? If an entry resolves an issue please reference the issue where applicable. You may suffix (it's optional) each entry with your name. Example of the format for the entry.
```
* Descriptive entry ([issue #123](https://github.com/PowerShell/xPSDesiredStateConfiguration/issues/123). [Name/Alias (@github_account)](https://github.com/github_account)
```

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/xscsma/23)*
<!-- Sent from Reviewable.io -->

Commit:
Updated Unreleased in Readme
